python/tank/bootstrap/manager.py

```diff
@@ -549,7 +549,7 @@ def _bootstrap_sgtk(self, engine_name, entity, progress_callback=None):
         tk = config.get_tk_instance(self._sg_user)
 
         if status != Configuration.LOCAL_CFG_UP_TO_DATE:
-            self._cache_apps(tk, progress_callback)
+            self._cache_apps(tk, engine_name, project_id, progress_callback)
```

Comment:
instead of just passing in the project here, i think we should pass in the full entity (e.g. project, shot, task or whatever it might be), resolve that into a context inside the method and pass that context to pick environment.

A pick environment hook may have business logic to trigger of say a task or something else, and in the case we filter that out (and only pass the project), we potentially are not getting the full business logic of the pick env hook.